lib/ash/actions/read/stream.ex

```diff
@@ -133,14 +133,14 @@ defmodule Ash.Actions.Read.Stream do
           {:halt, nil}
 
         offset ->
-          query =
+          {results, rest} =
             query
-            |> Ash.Query.limit(limit)
+            |> Ash.Query.limit(limit + 1)
             |> Ash.Query.offset(offset)
+            |> Ash.read!(Keyword.put(opts, :domain, domain))
+            |> Enum.split(limit)
 
-          results = Ash.read!(query, Keyword.put(opts, :domain, domain))
-
-          if Enum.count(results) == limit do
+          if Enum.empty?(rest) do
```

Comment:
Hmm....is there a way to accomplish this without having to read an additional record? For really large streams this means reading one additional record potentially many many times. Does `Enum.count(results) <= limit` not work?

Reply:
I went with this approach since it's the one used in pagination, but I guess there it needs to be like this to return the correct value for the `more?` field. I guess here we can read the right number and just do an extra empty read at the end.

Reply:
Actually, `Enum.count(results) <= limit` is always true by construction (the results will always be less than or equal to the limit). I guess the check would have to be `Enum.count(results) < limit`

Reply:
ah, yeah right :)